### DISCUSSION TITLE
URL length and what happens once the limit is reached.

### DISCUSSION THREAD

**Discussion by alteradeveloper** [YOURLS/YOURLS#2982] [Q&A] [answered]
Hi,

where do you change the length of the URL? It is currently configured on 4 random numbers only. Reached the limit (currently it would be 9999?) What happens?

Thanks in advance

**Comment by dgw** [comment, marked as answer]
I assume this is with the random-shorturls plugin. Adding a new link would enter an infinite loop.

You have more than just digits, though. At least lowercase letters, and perhaps also uppercase as well depending on YOURLS' config.

**Reply by alteradeveloper** [reply]
Yes with Random-shorturls. The infinite loop would cause the page to time out and thus error, right?

In order not to increase manually (for example from 4 to 5 numbers) is it possible to increase by 1 number once the limit is reached?

**Reply by ozh** [reply]
No.

**Reply by alteradeveloper** [reply]
OK thanks. A last question. By connecting YOURLS to a web application in which several users enter through a login, does the limit (in this case 9999 links, being configured to 4 numerical digits) apply to all links created regardless of the user who creates them? Or can it be configured so that each user has 9999 links? Thank you.

**Reply by ozh** [reply]
Sure. See https://github.com/YOURLS/YOURLS/wiki/Possible-With-A-Plugin

**Comment by rosano** [comment]
Outside of that plugin, is there a URL length limit? I'm hosting my own instance and would like to add very large links (over 10000 characters, content in the hash of the url…). It seems to just hang in the interface when doing so. In Chrome the console logs `net::ERR_CONNECTION_CLOSED`.

**Reply by dgw** [reply]
Browsers and servers both have de facto limits on URL length. Apache, for example, documents a [`LimitRequestLine` option](https://httpd.apache.org/docs/current/mod/core.html#limitrequestline) that may prevent processing very long GET requests such as those the YOURLS front-end sends when adding a new link. You can probably work around that by sending requests to YOURLS' API instead via POST.
